kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -150,8 +150,10 @@ private KaldbMetadataStoreChangeListener cacheNodeListener() {
 
   private void registerSearchMetadata(String snapshotName)
       throws ExecutionException, InterruptedException, TimeoutException {
+    // for "name" we pass url so that we can run two JVMs on the same host and not have znode path
+    // collision.
     this.searchMetadata =
-        new SearchMetadata(searchContext.hostname, snapshotName, searchContext.toUrl());
+        new SearchMetadata(searchContext.toUrl(), snapshotName, searchContext.toUrl());
```

Comment:
I think URL could contain protocol info going forward which can contain slashes that ZK doesn't like. So, it's better to use a different method to construct the node name from search method.

Reply:
I don't think the protocol stored anywhere today? 

If we were to store protocol in `kaldb_configs.proto` we would need to define how a query node can talk to an indexing/cache/recovery node 

If that sounds something we want we can do that in a separate PR

Reply:
fixed